src/server/services/database.service.ts

```diff
@@ -30,4 +31,28 @@ export class DatabaseService {
       );
     });
   }
+
+  async verifyUser(username: string, password: string): Promise<Boolean> {
+    return this.pool.connect((connection) => {
+      return connection.exists(
+        sql`SELECT * FROM users WHERE username = ${username} AND password_hash = ${password}`
+      );
+    });
+  }
+
+  async getUser(username: string): Promise<DatabaseUser | null> {
+    return this.pool.connect((connection) => {
+      return connection.maybeOne<DatabaseUser>(
+        sql`SELECT * FROM users WHERE username = ${username}`
+      );
+    });
+  }
+
+  async getPartialUser(username: string): Promise<User | null> {
+    return this.pool.connect((connection) => {
```

Comment:
why is this partialUser? It says it returns type 'User', which implicitly would mean its the full user